src/test/scala/za/co/absa/hyperdrive/trigger/scheduler/executors/shell/ShellExecutorTest.scala

```diff
@@ -0,0 +1,93 @@
+import za.co.absa.hyperdrive.trigger.models.enums.JobTypes.Shell
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration.Duration
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class ShellExecutorTest extends FlatSpec with Matchers with BeforeAndAfterAll with BeforeAndAfterEach with MockitoSugar {
+
+  private val updateJobStub: JobInstance => Future[Unit] = mock[JobInstance => Future[Unit]]
+
+  private val testScriptLocation = "/Users/abjba4j/Desktop/git/publicRepos/hyperdrive-trigger/src/test/resources/testShellScript.sh"
+  private val testJobInstance = JobInstance(
+    jobName = "jobName",
+    jobType = Shell,
+    jobParameters = JobParameters(variables = Map.empty[String, String], maps = Map.empty[String, Set[String]]),
+    jobStatus = InQueue,
+    executorJobId = None,
+    created = LocalDateTime.now(),
+    updated = None,
+    order = 0,
+    dagInstanceId = 0
+  )
+
+  override def beforeEach: Unit = {
+    org.mockito.Mockito.reset(updateJobStub)
+  }
+
+  "ShellExecutor.execute" should "succeeded job when everything is set correctly" in {
+    when(updateJobStub.apply(any[JobInstance])).thenReturn(Future.successful((): Unit))
+    val testInput = testJobInstance.copy(
+      jobParameters = testJobInstance.jobParameters.copy(
+        variables = Map[String, String](
+          "scriptLocation" -> testScriptLocation
+        )
+      )
+    )
+
+    Await.result(ShellExecutor.execute(testInput, updateJobStub.apply), Duration.Inf)
+
+    verify(updateJobStub, times(2)).apply(ArgumentMatchers.any())
+    verify(updateJobStub, times(1)).apply(ArgumentMatchers.eq(testInput.copy(jobStatus = Running)))
+    verify(updateJobStub, times(1)).apply(ArgumentMatchers.eq(testInput.copy(jobStatus = Succeeded)))
+  }
+
+  "ShellExecutor.execute" should "fail job when job with running status is executed" in {
+    when(updateJobStub.apply(any[JobInstance])).thenReturn(Future.successful((): Unit))
+    val testInput = testJobInstance.copy(jobStatus = Running)
+
+    Await.result(ShellExecutor.execute(testInput, updateJobStub.apply), Duration.Inf)
+
+    verify(updateJobStub).apply(ArgumentMatchers.eq(testInput.copy(jobStatus = Failed)))
+  }
+
+  "ShellExecutor.execute" should "fail job when script cant be found" in {
+    when(updateJobStub.apply(any[JobInstance])).thenReturn(Future.successful((): Unit))
+    val testInput = testJobInstance.copy(
+      jobParameters = testJobInstance.jobParameters.copy(
+        variables = Map[String, String]("scriptLocation" -> "/invalidLocation/invalidScriptName.sh")
+      )
+    )
+
+    Await.result(ShellExecutor.execute(testInput, updateJobStub.apply), Duration.Inf)
+
+    verify(updateJobStub, times(2)).apply(ArgumentMatchers.any())
+    verify(updateJobStub, times(1)).apply(ArgumentMatchers.eq(testInput.copy(jobStatus = Running)))
+    verify(updateJobStub, times(1)).apply(ArgumentMatchers.eq(testInput.copy(jobStatus = Failed)))
+  }
+
```

Comment:
Test for `jobStatus = Lost` is missing

Reply:
Status Lost is not applicable to this job type.

Reply:
I meant this line in `ShellExecutor`
```
      case _ => updateJob(jobInstance.copy(jobStatus = Lost))
```
It's not covered i.e. I can change `Lost` to another status and the tests still pass.

Reply:
Fixed.